### PR TITLE
Fixed error reported in issue #264

### PR DESCRIPTION
**Pull request by chedatomasz**
Also Added tests for the issue.
Regenerated bpr.cpp

### CODE REVIEW

**Comment by ita9naiwa** [comment]
please merge before commit.

**Comment by chedatomasz** [comment]
There is a bit of a mess with branches in my fork, so I opened a new PR with my fix merged with commits from master

**Comment by chedatomasz** [comment]
When you establish which of your changes you want to keep, please message me and I will prepare a fix for that version. Or you can do it yourself if you wish

**Comment by ita9naiwa** [comment]
@chedatomasz I'd like to handle this because this is caused because of me.

**Comment by chedatomasz** [comment]
I have rebased the branch onto the current master. Could you review and merge @ita9naiwa  ?

**Comment by chedatomasz** [comment]
@ita9naiwa ?

File: implicit/cuda/bpr.cu [hunk 1]
```diff
@@ -162,6 +162,7 @@ std::pair<int, int> bpr_update(const CudaVector<int> & userids,
     CHECK_CUDA(cudaFree(random_likes));
     CHECK_CUDA(cudaFree(random_dislikes));
     CHECK_CUDA(cudaFree(stats));
+    curandDestroyGenerator(rng);
```
Comment (benfred) [inline implicit/cuda/bpr.cu]: Nice catch! Thanks for fixing this

**Pull request merged**